server/game/gameSystems/DiscardCardsFromHand.ts

```diff
@@ -0,0 +1,102 @@
+import { CardTypeFilter, EventName, GameStateChangeRequired, Location, RelativePlayer, TargetMode } from '../core/Constants';
+import { AbilityContext } from '../core/ability/AbilityContext';
+import type Player from '../core/Player';
+import { IPlayerTargetSystemProperties, PlayerTargetSystem } from '../core/gameSystem/PlayerTargetSystem';
+import { Card } from '../core/card/Card';
+import { DiscardSpecificCardSystem } from './DiscardSpecificCardSystem';
+import { cardTypeMatches } from '../core/utils/EnumHelpers';
+import { asArray } from '../core/utils/Helpers';
+import { assertTrue } from '../core/utils/Contract';
+
+export interface IDiscardCardsFromHandProperties extends IPlayerTargetSystemProperties {
+    amount: number;
+
+    /* TODO: Add reveal system to when card type filter or card condition exists, as this is required to keep the
+    in order to keep the player honest in a in-person game */
+    cardTypeFilter?: CardTypeFilter | CardTypeFilter[];
+    cardCondition?: (card: Card, context: AbilityContext) => boolean;
+}
+
+export class DiscardCardsFromHand<TContext extends AbilityContext = AbilityContext> extends PlayerTargetSystem<TContext, IDiscardCardsFromHandProperties> {
+    protected override defaultProperties: IDiscardCardsFromHandProperties = {
+        amount: 1,
+        cardCondition: () => true
+    };
+
+    public override name = 'discard';
+    public override eventName = EventName.OnCardsDiscardedFromHand;
```

Comment:
I think we should make sure that the event names for this and `DiscardSpecificCardSystem` are really clear in terms what the distinction is. I think this one is currently good, but we might want to change the specific card event to `OnCardDiscarded` singular